app/services/gitlab/fetch_branches_service.rb

```diff
@@ -0,0 +1,33 @@
+module Gitlab
+  class FetchBranchesService
+    delegate :error_message, to: :api, allow_nil: true
+
+    def initialize(project)
+      @project = project
+    end
+
+    def fetch
+      @api ||= GitlabApi::FetchBranches.perform(
+        gitlab_integration.data['url'],
+        gitlab_integration.data['token'],
+        gitlab_integration.data['project_id']
+      )
+
+      api.success?
+    end
+
+    def branches
+      return unless api&.success?
```

Comment:
why the &. here? shouldn't have an api here?

Reply:
jus saw the api class you don't  need the `&.`

Reply:
If branches is called before fetch, it'll throw an error since api will be null. That's why I did it this way.

Reply:
why the need to call fetch why can't the developer just call branches?